src/main/java/seedu/uninurse/model/person/PatientMatchPredicate.java

```diff
@@ -0,0 +1,90 @@
+package seedu.uninurse.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.uninurse.model.condition.ConditionContainsKeywordsPredicate;
+import seedu.uninurse.model.tag.TagContainsKeywordsPredicate;
+import seedu.uninurse.model.task.TaskContainsKeywordsPredicate;
+
+/**
+ * Tests that a {@code Patient} matches with all descriptors.
+ */
+public class PatientMatchPredicate implements Predicate<Patient> {
+    private final PatientContainsKeywordsPredicate patientContainsKeywordsPredicate;
+    private final NameContainsKeywordsPredicate nameContainsKeywordsPredicate;
+    private final PhoneContainsKeywordsPredicate phoneContainsKeywordsPredicate;
+    private final EmailContainsKeywordsPredicate emailContainsKeywordsPredicate;
+    private final AddressContainsKeywordsPredicate addressContainsKeywordsPredicate;
+    private final TagContainsKeywordsPredicate tagContainsKeywordsPredicate;
+    private final ConditionContainsKeywordsPredicate conditionContainsKeywordsPredicate;
+    private final TaskContainsKeywordsPredicate taskContainsKeywordsPredicate;
```

Comment:
I think you can just have a `List<Predicate<Patient>>` or maybe a `Map<Predicate<Patient>>` instead.

Reply:
Done, used `List<Predicate>`